rtengine/diagonalcurves.cc

```diff
@@ -106,18 +106,18 @@ DiagonalCurve::DiagonalCurve (const std::vector<double>& p, int poly_pn)
                 }
 
                 for (int i = 4; i < 8; i++) {
-                    x[i] = (p[i] + 100.0) / 200.0;
+                    x[i] = max((p[i] + 100.0) / 200.0, 0.001);
                 }
 
                 if (p.size() < 9) {
-                    x[8] = 1.0;
+                    x[8] = 0.99;
                 } else {
-                    x[8] = p[8] / 100.0;
+                    x[8] = min(p[8] / 100.0, 0.99);
```

Comment:
Clamping these values results in a very small but imperceptible change in the output. Is it possible to come up with an alternative solution, or use values closer to the original to minimize the difference?

Reply:
Same comment for backwards compatibility.
I also cannot find where the eighth parameter is actually used in the GUI, in my quick tests it was always 1.0.

Reply:
The 8th parameter controls the smoothness of the curve (it becomes "blocky" if turned down), and seems to be fixed indeed. When set to 1.0 `pfull` starts to have some issue (returns 0.0 regardless of t) and thus one would have to look at this function to accommodate that value.